samples/dnn/speech_recognition.py

```diff
@@ -0,0 +1,455 @@
+        y = np.pad(y, int(n_fft // 2), mode=pad_mode)
+
+        # Window the time series.
+        y_frames = self.frame(y, frame_length=n_fft, hop_length=hop_length)
+
+        # Convert data type to complex
+        dtype = self.dtype_r2c(y.dtype)
+
+        # Pre-allocate the STFT matrix
+        stft_matrix = np.empty( (int(1 + n_fft // 2), y_frames.shape[-1]), dtype=dtype, order="F")
+
+        stft_matrix = np.fft.rfft( fft_window * y_frames, axis=0)
+        return stft_matrix if return_complex==True else np.stack((stft_matrix.real,stft_matrix.imag),axis=-1)
+
+class Decoder():
+    '''
+        Used for decoding the output of jasper model.
+    '''
+    def __init__(self):
+        labels=[' ','a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z',"'"]
+        self.labels_map = {i: label for i,label in enumerate(labels)}
+        self.blank_id = 28
+
+    def decode(self,x):
+        """
+            Takes output of Jasper model and performs ctc decoding algorithm to
+            remove duplicates and special symbol. Returns prediction
+        """
+        x = np.argmax(x,axis=-1)
+        hypotheses = []
+        prediction = x.tolist()
+        # CTC decoding procedure
+        decoded_prediction = []
+        previous = self.blank_id
+        for p in prediction:
+            if (p != previous or previous == self.blank_id) and p != self.blank_id:
+                decoded_prediction.append(p)
+            previous = p
+        hypothesis = ''.join([self.labels_map[c] for c in decoded_prediction])
+        hypotheses.append(hypothesis)
+        return hypotheses
+
+if __name__ == '__main__':
+
+    # Computation backends supported by layers
+    backends = (cv.dnn.DNN_BACKEND_DEFAULT, cv.dnn.DNN_BACKEND_HALIDE, cv.dnn.DNN_BACKEND_INFERENCE_ENGINE, cv.dnn.DNN_BACKEND_OPENCV)
+    # Target Devices for computation
+    targets = (cv.dnn.DNN_TARGET_CPU, cv.dnn.DNN_TARGET_OPENCL, cv.dnn.DNN_TARGET_OPENCL_FP16)
+
+    parser = argparse.ArgumentParser(description='This script runs Jasper Speech recognition model',
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--input_audio', type=str, help='Path to input audio file.')
+    parser.add_argument('--show_spectrogram', action='store_true', help='Whether to show a spectrogram of the input audio.')
+    parser.add_argument('--model', type=str, default='jasper.onnx', help='Path to the onnx file of Jasper. default="jasper.onnx"')
+    parser.add_argument('--output', type=str, help='Path to file where recognized audio transcript must be saved. Leave this to print on console.')
+    parser.add_argument('--backend', choices=backends, default=cv.dnn.DNN_BACKEND_DEFAULT, type=int,
+                        help='Select a computation backend: '
+                        "%d: automatically (by default) "
+                        "%d: Halide"
```

Comment:
I'm not sure that Halide backend supports Jasper

Reply:
I forgot to remove this. I was checking if it does.